linstor.go

```diff
@@ -973,3 +973,28 @@ func ValidResourceName(resName string) error {
 
 	return nil
 }
+
+// LinstorifyResourceName tries to generate a valid LINSTOR name if the input currently is not.
+// If the input is already valid, it just returns this name.
+// This tries to preserve the original meaning as close as possible, but does not try extra hard.
+// Do *not* expect this function to be injective.
+// Do *not* expect this function to be stable. This means you need to save the output, the output of the function might change without notice.
+func LinstorifyResourceName(name string) (string, error) {
```

Comment:
It would be really nice to have a test for this, if not just to provide some "goofus and gallant" style examples of what transformations are going to be done, especially if more are added moving forward.

Reply:
True. Tests are always good :)